### DISCUSSION TITLE
Support variables in "Filter by value" transformation

### DISCUSSION THREAD

**Discussion by Alex389-r** [grafana/grafana#52219] [Dashboards, Panels & Visualizations]
Hello,

I have a dashboard where I am calling a REST API to print some data on screen. I am in need of duplicate this data per "X" variable called "Test_Variable". I have tried repetaing the panels or the rows and obviusly is working but is making lots of calls to my REST API so I cannot afford it.

To solve this, I have created a single table tha contains all the data i need and the idea is to make my entire dashboard reusing the data from the table, so Im not making tons of calls. 

![image](https://user-images.githubusercontent.com/109079901/178960785-dc670927-2a02-448d-89cf-29b8c6dfea6e.png)

The problem here is that I need my panels to give me only the data from one row each time (Based on my variable "Test_Variable"). To achieve this, im using the transformation "Filter data by values" so I am filtering the data to show based on 1 column value of the table. After that im selecting which data to show:

![image](https://user-images.githubusercontent.com/109079901/178961335-83538495-304d-4657-8982-a9674049249e.png)
![image](https://user-images.githubusercontent.com/109079901/178961362-98a198c9-e182-44bc-a464-f07e84db9fcf.png)
![image](https://user-images.githubusercontent.com/109079901/178961394-e4a872f7-3d01-42ee-a839-affa3430d02b.png)
![image](https://user-images.githubusercontent.com/109079901/178961433-15bb473e-09db-4ca0-9c57-694dd9a764a5.png)

As you see this is working correctly, but now I want my row to be repeated X times based on my variable and I cannot achieve it because it seems that this transformation "Filter data by value", does not support the use of variables

![image](https://user-images.githubusercontent.com/109079901/178961916-0d70d5f1-5077-43d7-9bd1-451c803060b6.png)

Of course i need this to be "smart" so if my row is repeated based on the same variable, the frist row gives me the first value, the second row, the second value... etc. Basically same as using it in the panel title, for example:

![image](https://user-images.githubusercontent.com/109079901/178962366-0aab4015-2fb5-4b5c-9cbd-6e128a5c8fe1.png)

Right now my only solution is to create the panels manually and adjust the value manually to each one but this would be so easy with the use of the variables on this transformation (And other ones too).

I have been looking for this in all the requests and I have found the following:

- https://github.com/grafana/grafana/issues/25469 --> Someone asking more less the same as me and it apparently was closed but the variables only works in "Add field from calculation"

- https://github.com/grafana/grafana/pull/44486 --> This is the solution for the previous request. But as I said it doesnt solves the core problem that is supporting variables in ALL transformations (Not only in "Add field from calculation").

- https://github.com/grafana/grafana/issues/42267 --> Another user asking support for supporting variables in "Filter by name". This case was closed but it was not really solved. 

- https://github.com/grafana/grafana/pull/51804 --> This is the solution that he asked for, but only for the "Filter by name" transformation. Right now is not closed.

- https://github.com/grafana/grafana/discussions/50451 --> Another user asking for the adding variable support to "Filter by name" transformation

**As a conclusion:** Right now the variables are only working in some cases ("Add field from calculation" transformation). The variables doesnt seem to work in other transformations so this why some people keeps asking for this.

In my case, I would love this feature to be added to "Filter by value" transformation but of course it would be wonderful if this could be made standar across all of the transformation.

Thank you

**Comment by ShaunRoselt** [comment]
[Variable not working within transform rename](https://github.com/grafana/grafana/issues/61239) #61239

I agree. This would be such a useful feature to have. 

In my case, I need it for the rename transform as I want to rename things from a variable value.

**Comment by ShaunRoselt** [comment]
Message from @gtk-grafana:

> It looks like this would require migrating all transformations to core, so I would take a guess this isn't coming out any time soon unless metrics de-prioritizes work on other focus areas (i.e. prometheus, graphite, influx)

**Reply by ShaunRoselt** [reply]
Looks like it's possibly planned for v10: 

#46428
#55605

**Comment by gelicia** [comment]
Hello! We want to let you all know that we are moving to Github Issues for feature requests. You can read more about the move [here](https://github.com/grafana/grafana/discussions/73424).

This feature request seems to be covered mainly by https://github.com/grafana/grafana/issues/46428 , so I will be closing this discussion so conversation about this feature can continue there.
